### ISSUE TITLE
Making sails-docs multilingual(2)

### ISSUE THREAD

**Issue by kory-yhg** [balderdashy/sails-docs#580]
I made a new topic for this issue, since #149 has already closed.

I finished translation for concept, anatomy and getting-started sections.
(See #576 and #579)
Also I made  the documents up-to-date. Fortunately, the document is quite stable so it took only 3 days to follow the update for one year.
Now we have decent amount of documents for people starting sails in Japan.

I think it going to be useful if we have one in the official website as well.
Looking for former discussion, it would be better we have a combo-box for choosing language, and making uri for for each language, e.g:
http://sailsjs.org/#/documentation/ -> English docs (Default)
http://sailsjs.org/#/ja/documentation/ -> Japanese Docs

Or if it takes so long to develop some code for it, we can simply deploy same code with another branch and domain.
I have sailsjs.jp and happy to prepare some instances if someone tell me how to do it.

**Comment by brettimus** [comment]
Ping me if this doesn't get merged in the next few days! Would like time to understand how localization is handled on the site.

**If another maintainer is comfortable merging, please do so!**

**Comment by brettimus** [comment]
Let's close it. 🙏

**Comment by kory-yhg** [comment]
Please not close it. The topic about this issue is to make http://sailsjs.org into multilingual, which is not finished yet.

**Comment by kory-yhg** [comment]
I'm wondering if we can make @mikermcneil on board to this topic. I guess he knows something about it.

**Comment by brettimus** [comment]
Reopened!

**Comment by sailsbot** [comment]
Thanks for posting, @kory-yhg.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by kory-yhg** [comment]
Hi, guys. I launched one for Japanese.
http://sailsjs.jp/
We can access about 86% of Sails documentation translated into Japanese now.
But still have to think about how we can make it multilingual since I just forked the original and deployed it.

**Comment by mikermcneil** [comment]
@kory-yhg this is amazing, great work!  Thanks so much for your help!

To your point about ongoing multilingual deployments, I _think_ it might be as straightforward as having doctemplater pull from the appropriate branch [here](https://github.com/balderdashy/www.sailsjs.org/blob/master/api/machines/_project_3549_0.0.43/Compilemarkdowndocs.js#L166).  

As far as the broader multilingual strategy, I think it would work to have more specific branches for specific versions, e.g. `ja-0.11.3`.  The reason we started doing branches like `0.11.3` was to allow us to merge PRs in sails-docs without necessarily having the changes go live (i.e. in case they apply to a change that is on master and came from a PR in Sails or Waterline core).

That said, I think the _best_ approach would actually be to stick with a `ja` branch, but then create tags like `ja-0.11.3` from it (and do the same thing for the default English site on `master` or `en`).  Unfortunately, pulling from tags is not supported in doctemplater yet.  It would probably be pretty straightforward to change though (will follow up)  In the mean time, I think it's no problem to go with the assumption that the `ja` branch is translated from the latest stable content.

Thanks again Kory, and great work!

**Comment by mikermcneil** [comment]
(the "bug" label is just to ward off sailsbot)

**Comment by mikermcneil** [comment]
Here's a link to doc-templater:  https://github.com/uncletammy/doc-templater

**Comment by mikermcneil** [comment]
btw thanks @brettimus.  Sorry about the slow reply- been in the thick of things on the Treeline and Sails book front.  Assuming you, @rachaelshaw, and @kory-yhg are all on board for this approach, I'm happy to help get doc-templater working w/ tags.  The only other thing is that we can't deploy to Heroku automatically using tags, but we can work around that (I'd rather have to do it on the CLI or take a few extra steps and still be able to take advantage the cleaner approach).

@kory-yhg as far as being able to navigate between versions, I really like your idea of a dropdown that lets you pick between available languages.  At the bottom of that dropdown, we could include a blurb along the lines of "Don't see your language here? Click here to help translate the Sails docs.".  As far as the URL structure, we could just hard-code the destination URL in the HTML in this repo (I really like `sailsjs.ja` if you don't mind volunteering to keep the domain and the content hosted there-- if it ever becomes a burden, just let us know).

**Comment by mikermcneil** [comment]
@kory-yhg @brettimus http://asksailsbot.tumblr.com/post/136824812101/new-years-resolutions-of-github-issues

**Comment by mikermcneil** [comment]
(oops, didn't mean to close this until we hear back from Kory)

**Comment by kory-yhg** [comment]
Sorry for late on reply, I'm happy to host sailsjs.jp.
And finally finished translating all the documents. (by pull request #630)

**Comment by mikermcneil** [comment]
@kory-yhg great work! :clap: 

And @all:  new process for translations is in the README of this repo: https://github.com/balderdashy/sails-docs/blob/master/README.md

**Comment by ChinuxParibus** [comment]
It seems that the "es" documentation branch has abandoned, so i can see the last commit is from *Sep, 29. 2014*. How can i collaborate? Just PR? I'm really interested on translate in spanish...

**Comment by irlnathan** [comment]
Howdy @ChinuxParibus, a PR would be great!

**Comment by ChinuxParibus** [comment]
Good, i'm on it... :+1:
